test-harness/specs/parameters-ac1.oas2.txt

```diff
@@ -0,0 +1,28 @@
+====test====
+When I send a request to an operation
+And the operation has a required path segment
+And in the request I sent that segment is missing
+I should get a 400 response back
+And the payload of the response should be in application/problem+json content-type
+====spec====
+swagger: '2.0'
+paths:
+  /pet/a/{b}:
+    get:
+      parameters:
+        - in: path
+          name: b
+          required: true
+          type: number
+      responses:
+        '200':
+          description: ok
+          schema:
+            type: string
+====server====
+mock -p 4010
+====command====
+curl -sIXGET http://127.0.0.1:4010/pet/a
+====expect====
+HTTP/1.1 400 Bad Request
```

Comment:
thinking out-loud here: if the path segment is required, and it is not provided in the request, does it still make the request correct syntax wise? Not so sure. Or maybe it should be 404?

Reply:
> @philsturgeon I guess the reason we're not returning 422 is because this is not related to the payload, but to the query/headers, right?

This is a false distinction I think that is caused by OpenAPI treating a lot of different types of thing as a parameter. Path parameters are part of the path, and if you enter a path which does not exist, you should get a 404. 

So, if you request `/a` but the path was expecting `/a/b` then that should be a 404. 

If you request `/a/foo` but the operation parameters have validation rules saying foo it should have been `format: uuid`, then that should either be 422 (validation failure) or 404 (you entered a nonsense path), but there is nothing here that relates to syntax or 400.

Reply:
@philsturgeon So if you look at this specific test — 404 is the status code that should be returned, thous the harness expectation is wrong. Is that correct?

Reply:
404 if the URL is missing segments because that would be a non existent URL and that’s what 404 is all about. 

If for now everything is doing 422 that’s ok for now. I’m not sure where the 400 came from that’s not right.

Reply:
It's not always 422 — we'll get 404 in cause of a missing path parameter, because it's an error thrown from the router component, which only emits 404 at the moment.